### PR TITLE
Don't Merge: This is a demo for vercel

### PR DESCRIPTION
**Pull request by Xuanwo**

### CODE REVIEW

**Comment by Xuanwo** [comment]
ping @PsiACE for a look for the vercel's native PR preview support.

**Comment by PsiACE** [comment]
wow, love it

**Comment by ZhiHanZ** [comment]
seems out of topic... Vercel seems to be an interesting tool for frontend prototyping, what is the difference between vercel and netlify github action?

**Comment by Xuanwo** [comment]
@ZhiHanZ 

There are two things here: Website hosting and PR preview. Both vercel, netlify and gh-pages all provide support for website hosting. However, not all platform has support for per-PR preview.

- Vercel has native support (as in this PR)
- Netlify didn't have native support ([but now they have](https://www.netlify.com/products/deploy-previews/)), so the community built an action [Netlify Actions](https://github.com/marketplace/actions/netlify-actions) to fix it.
- Github Pages don't have a PR preview and it's hard to implement (because GitHub only serve content from the `gh-pages` branch).

I will start a discussion about all possible plans so that we can come a proposal to decide how we deploy our site. I expect to finish the job during stage 2 as described in #3210.

**Comment by ZhiHanZ** [comment]
> @ZhiHanZ
> 
> TL;DR
> 
> They function similarly and both work.
> 
> There are two things here: Website hosting and PR preview. Both vercel, netlify and gh-pages all provide support for website hosting. However, not all platform has support for per-PR preview.
> 
> * Vercel has native support (as in this PR)
> * Netlify didn't have native support ([but now they have](https://www.netlify.com/products/deploy-previews/)), so the community built an action [Netlify Actions](https://github.com/marketplace/actions/netlify-actions) to fix it.
> * Github Pages don't have a PR preview and it's hard to implement (because GitHub only serve content from the `gh-pages` branch).
> 
> I will start a discussion about all possible plans so that we can come a proposal to decide how we deploy our site. I expect to finish the job during stage 2 as described in #3209.

Thanks for clarification! on possible improvement on vercel is to make PR preview updates only when files under /websites got changed.

**Comment by Xuanwo** [comment]
> one possible improvement on vercel is to make PR preview updates only when files under /websites got changed, maybe we could incorporate it into tracking issue

Makes sense, updated.

**Comment by Xuanwo** [comment]
> one possible improvement on vercel is to make PR preview updates only when files under /websites got changed, maybe we could incorporate it into tracking issue

Implemented, it's a vercel builtin feature.

image

**Comment by sundy-li** [comment]
Great tool, IMO `vercel` bot can be disabled if nothing changes related to the doc.

**Comment by Xuanwo** [comment]
> Great tool, IMO `vercel` bot can be disabled if nothing changes related to the doc.

Oh, I need to figure it out. There is no obvious way to do that.
